[[ Bug 22060 ]] Stop point values with matching hashes from overwriting each other.

livecodeian opened the pull request:
False positives in MCCanvasPoint compare function were leading to point
values with duplicate hashes being incorrectly identified as unique,
causing the most recent value to be replaced with the existing value.

This patch fixes the comparison function in order to remove these false
positives.

runrevmark commented:
@livecodeian Is this a bug present in 9.0.x? Or is it non-user-visible (i.e. won't have any effect fixing it on that branch)?

livecodeian committed:
[[ Bug 22060 ]] Stop point values with matching hashes from overwriting each other.

False positives in MCCanvasPoint compare function were leading to point
values with duplicate hashes being incorrectly identified as unique,
causing the most recent value to be replaced with the existing value.

This patch fixes the comparison function in order to remove these false
positives.

livecodeian commented:
The bug affects 9.0 versions so I've rebased the branch.

montegoulding commented on docs/lcb/notes/22060.md:
```diff
@@ -0,0 +1,4 @@
+# LiveCode Builder Host Library
+## Canvas library
+
+# Fix bug in canvas library that can result in Points being initialized with previously used values.
```
@livecodeian I think there is an errant `#` here which will stuff up the release notes parsing

montegoulding replied:
@livecodeian oh I think you may just want to make it like this so there's just a bugfix note
```
# [22060] Fix bug in canvas library that can result in Points being initialized with previously used values.
```

montegoulding replied:
@livecodeian I'm fairly sure it needs the `[22060]` bit to figure out the bug number

runrevmark commented:
@livecode-vulcan review ok 39a327d

livecode-vulcan commented:
💙 review by @runrevmark ok 39a327d2a1d90c64dc3dc6e1494174ca8a833c8a

livecode-vulcan commented:
😎 test success 39a327d2a1d90c64dc3dc6e1494174ca8a833c8a

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/527)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/519)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/519)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/519)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/519)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2041)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/269)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/268)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2073)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2063)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2061)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2096)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2033)
